packages/rocketchat-lib/server/sendMessage.coffee

```diff
@@ -116,9 +119,20 @@ RocketChat.sendMessage = (user, message, room, options) ->
 					query._id =
 						$in: mentionIds
 
-				usersOfMentionIds = _.pluck(usersOfMention, '_id');
-				if usersOfMentionIds.length > 0
-					for usersOfMentionId in usersOfMentionIds
+				userIdsToNotify = _.pluck(usersOfMention, '_id')
+
+				# If the message is @all, send a notification to all online room users except for the sender.
+				if toAll and room.usernames?.length > 0
+					onlineUsersOfRoom = RocketChat.models.Users.find({
+							username: {$in: room.usernames},
+							_id: {$ne: user._id}
+							status: {$in: ['online', 'away', 'busy']}},
```

Comment:
I don't agree with this, we should notify the browsers of all online users but we should notify all non online users via push notification, don't you think?